File: backend/data/importdata.js
const fs = require("fs")
const GD = require("gd.js");

const gd = new GD();

async function getDemonList(amount) {
    let res = await fetch("https://pointercrate.com/api/v2/demons/listed?limit=" + amount)
    let data = await res.json()
    return data
}

async function getLevel(id) {
    let level = await gd.levels.get(id)
    return level
}


let finalData = []

async function transform() {
    let demons = await getDemonList(100)
    for(let i=0; i<demons.length; i++){
        let data = await getLevel(demons[i]["level_id"])
        let transformed = {"id": demons[i]["level_id"], "name": demons[i]["name"],"position": demons[i]["position"], "creator": demons[i]["publisher"]["name"], "verifier": demons[i]["verifier"]["name"], "thumbnail": demons[i]["thumbnail"], "downloads": data["stats"]["downloads"], "length": data["stats"]["length"]["raw"]}
        finalData.push(transformed)

    fs.writeFile("finaldata.json", JSON.stringify(finalData, null, 2), (err) => {
    if (err){
        console.log("error writing file")
    }
    else{
        console.log("writing successful")
    }
})
}
    
}

transform()

File: backend/data/importdata.py
import requests, json

def fetch_api_data(url):
    try:
        # Send a GET request to the API endpoint
        response = requests.get(url)

        # Check if the request was successful (status code 200)
        if response.status_code == 200:
            print("Request successful!")
            # Parse the JSON response into a Python dictionary
            data = response.json()
            return data
        else:
            print(f"Request failed with status code: {response.status_code}")
            return None
    except requests.exceptions.RequestException as e:
        print(f"An error occurred: {e}")
        return None

if __name__=="__main__":
    with open("demonlist.json", "w") as f:
        output = {}
        demonlist = fetch_api_data("https://pointercrate.com/api/v2/demons/listed?limit=100")
        for i in demonlist:
            leveldata = fetch_api_data("/api/level/" + i["level_id"])
            id = leveldata["Meta"]["ID"]
            output[i["name"]] = leveldata["Meta"]["Name"]
            output[i["id"]] = id
            output[i["position"]] = i["position"]
            output[i["creator"]] = leveldata["Publisher"]["name"]
            output[i["verifyer"]] = leveldata["Meta"]["Name"]
            output[i["thumbnail"]] = i["thumbnail"]
            output[i["downloads"]] = leveldata["Meta"]["Name"]
            output[i["length"]] = leveldata["Meta"]["Length"]


